### PR TITLE
control #collections for support bundle

### PR DESCRIPTION
**Pull request by gz**
## Checklist

- [x] Unit tests added/updated
- [x] Integration tests added/updated
- [x] Documentation updated
- [ ] Changelog updated

## Breaking Changes?

No

### CODE REVIEW

**Review by ryzhyk — APPROVED** [review]

File: crates/pipeline-manager/src/api/endpoints/pipeline_interaction/support_bundle.rs [hunk 1]
```diff
@@ -16,6 +18,68 @@ use actix_web::{
...
+    #[serde(flatten)]
+    summary: CollectionSummary,
+}
+
+const METADATA_VERSION: u32 = 1;
+
+/// Format `DateTime<Utc>` for use as a directory name. Colons are problematic
+/// on some filesystems and inside ZIP tooling, so we keep the rfc3339 layout
+/// but swap `:` for `-` (e.g., `2026-05-25T12-34-56.789Z`).
+fn timestamp_dir(time: &DateTime<Utc>) -> String {
```
Comment (mythical-fred) [inline crates/pipeline-manager/src/api/endpoints/pipeline_interaction/support_bundle.rs:62]: Docstring example claims `2026-05-25T12-34-56.789Z`, but `chrono::DateTime<Utc>::to_rfc3339()` emits `+00:00`, so after `.replace(':', '-')` the directory name is actually `2026-05-25T12-34-56.789+00-00`. Either fix the comment or switch to a `Utc.format("%Y-%m-%dT%H-%M-%S%.fZ")` style format for the cleaner shape the comment implies.

**Review by mythical-fred — COMMENTED** [review]
Bundle layout cleanup is a clear win — per-collection directory + structured `metadata.json` is a much better foundation than the `<ts>_<file>` flattening + plain manifest. The data-version bump from 4 → 5 is handled cleanly (Helper struct + `deserialize_v4_payload_marks_pipeline_events_missing` proves the round-trip), and the new tests (`bundle_layout_groups_files_by_timestamp_directory`, `duplicate_collection_timestamps_get_suffixed_directories`, `test_support_bundle_limit`) cover the core happy paths.

Not blocking, but a few things worth tightening before this lands:

1. **PR scope / description out of date.** The title says *control #collections for support bundle*, but commits 2-5 are unrelated piggybacks: `02fc902` is a real Python SDK bugfix (`FelderaClient.approve_pipeline` was missing the `silent_bootstrap` kwarg that `Pipeline.approve` was already passing — so today calling `pipeline.approve(silent_bootstrap=True)` raises `TypeError`), `487c3e4` defers deltalake imports, and `76bf8bd` bumps deltalake 1.5 → 1.6 and drops every `@skip_on_arm64`. Either split them out (the silent-bootstrap fix is small + isolated, would be trivial), or at minimum spell them out in the PR description so the changelog reflects reality. The "Documentation updated" checkbox is also ticked but I don't see any `docs/**` changes — the new `limit` query param + `--limit`/`-n` fda flag + `pipeline_events` would normally land alongside a docs note.

2. **`timestamp_dir`'s docstring claims the format ends in `Z`** (`2026-05-25T12-34-56.789Z`), but `chrono::DateTime<Utc>::to_rfc3339()` emits `+00:00`, so after `.replace(':', '-')` the directory name actually ends in `+00-00`. Fine as a unique key, but the comment lies. Use `format_with_items` (or `Utc.format("%Y-%m-%dT%H-%M-%S%.fZ")`) if you want the cleaner `Z` form everyone reads.

3. **Tests don't pin the `limit=0 → 400` path.** That branch is half the point of `InvalidSupportBundleParameter` and the new `400` arm in the OpenAPI doc. A quick handler-level unit test (or pytest hitting the server) would lock it in. Same for `limit + collect=false` — the new python `test_support_bundle_limit` only exercises `collect=True`, which is the easy direction.

4. **`silent_bootstrap` SDK fix has no test.** `feldera_client.approve_pipeline(name, silent_bootstrap=True)` will round-trip through the new `params` dict, but nothing in this PR asserts that. Given the bug it fixes was "public method silently broken," a one-line `test_shared_pipeline` style check (`approve` accepts the kwarg and forwards `?silent_bootstrap=true`) would protect it.

5. **`metadata_version: 1` is great forward-thinking, but `fda debug unbundle` doesn't read it.** When you bump to 2, old `fda` binaries will silently parse new bundles in undefined ways. Either log + warn in `unbundle_support_bundle` when `metadata_version > known`, or note in code that consumers MUST check.

6. **fda's `rsplit_once('/').map(|(dir, _)| dir).unwrap_or("unknown")` in `unbundle_support_bundle`** means a pipeline-config file at the ZIP root (older bundle, or a hand-edited one) silently gets `unbundled_unknown` as the suffix and could collide across runs. Not breaking, but a warning-and-skip would be safer than munging.

7. **`render_metadata_text` mixes `✓ ✗ ⚠` glyphs into `metadata.txt`.** Customers' support pipelines that grep these (or open in cp1252 viewers) will hit mojibake. Plain `OK:` / `FAIL:` / `SKIP:` would survive every channel; the JSON version already carries the structured form for tooling.

Nothing here blocks — ryzhyk already approved and the design is sound. Just hygiene worth tightening, particularly (1).

File: crates/pipeline-manager/src/api/endpoints/pipeline_interaction/support_bundle.rs [hunk 2]
```diff
@@ -160,6 +293,11 @@ pub(crate) async fn get_pipeline_support_bundle(
 ) -> Result<HttpResponse, ManagerError> {
     let pipeline_name = path.into_inner();
     let support_bundle_params = query.into_inner();
+    if support_bundle_params.limit == Some(0) {
+        return Err(ManagerError::from(ApiError::InvalidSupportBundleParameter {
+            reason: "limit must be at least 1 when provided; omit it to return all retained collections".to_string(),
+        }));
+    }
```
Comment (mythical-fred) [inline crates/pipeline-manager/src/api/endpoints/pipeline_interaction/support_bundle.rs:300]: The new `400` arm is the headline of this commit but it has no test coverage. A quick handler-level test calling `get_pipeline_support_bundle` with `limit=Some(0)` and asserting the `InvalidSupportBundleParameter` error would lock it in alongside the existing `bundle_layout_*` tests.

File: python/tests/platform/test_shared_pipeline.py [hunk 3]
```diff
@@ -780,8 +783,24 @@ def test_support_bundle_with_selectors(self):
...
+
+        support_bundle_bytes = self.pipeline.support_bundle(limit=1)
+        assert isinstance(support_bundle_bytes, bytes)
+
+        with zipfile.ZipFile(io.BytesIO(support_bundle_bytes), "r") as zip_file:
+            file_list = zip_file.namelist()
+            collection_dirs = {
+                item.split("/", 1)[0] for item in file_list if "/" in item
+            }
+            assert len(collection_dirs) == 1
```
Comment (mythical-fred) [inline python/tests/platform/test_shared_pipeline.py:803]: Only exercises `limit=1` with `collect=True`. The other half of the design — `limit=N` + `collect=False` returning the N most recent **previously stored** collections — is unverified. A two-step test (start pipeline, sleep until at least two retained collections exist, call `support_bundle(collect=False, limit=1)`, assert exactly one directory and that it's the newest) would cover the case the docstring sells most aggressively.

File: python/feldera/rest/feldera_client.py [hunk 4]
```diff
@@ -672,9 +672,22 @@ def pause_pipeline(
...
     ):
+        """
+        Approve a pipeline awaiting approval to proceed with bootstrapping.
+
+        :param pipeline_name: The name of the pipeline to approve.
+        :param silent_bootstrap: Set True to bootstrap with output connectors
+            disabled, so no records are emitted during the bootstrap phase.
+            False by default.
+        """
+
```
Comment (mythical-fred) [inline python/feldera/rest/feldera_client.py:685]: This is actually a behavior fix: on `main`, `Pipeline.approve(silent_bootstrap=True)` already passed the kwarg here but `approve_pipeline` didn't accept it, so the public method raised `TypeError`. Worth (a) mentioning in the PR description so the changelog catches it, and (b) adding a tiny test that exercises `approve_pipeline(name, silent_bootstrap=True)` end-to-end — currently nothing pins the fix.

File: crates/fda/src/debug.rs [hunk 5]
```diff
@@ -245,10 +248,11 @@ async fn unbundle_support_bundle(
         let name = config["name"]
             .as_str()
             .ok_or_else(|| anyhow::anyhow!("Pipeline name not found in config"))?;
-        // Extract timestamp from config file name to make pipeline name unique
+        // The directory name carries the collection timestamp, which we mix
+        // into the new pipeline name to keep it unique across collections.
         let timestamp = config_file
-            .split('_')
-            .next()
+            .rsplit_once('/')
```
Comment (mythical-fred) [inline crates/fda/src/debug.rs:254]: If an older bundle (or one hand-edited) ever has `pipeline_config.json` at the ZIP root, this `unwrap_or("unknown")` collapses every such entry to `..._unbundled_unknown`, silently colliding across runs. Cheap defensive option: skip with a `warn!` when no leading directory is present, since post-#6319 bundles always have one.

File: crates/pipeline-manager/src/api/endpoints/pipeline_interaction/support_bundle.rs [hunk 6]
```diff
@@ -16,6 +18,68 @@ use actix_web::{
...
+    /// Directory inside the ZIP that holds this collection's files.
+    directory: String,
+    time: DateTime<Utc>,
+    /// Version of the support bundle data format that produced this entry.
+    data_version: u32,
+    #[serde(flatten)]
+    summary: CollectionSummary,
+}
+
+const METADATA_VERSION: u32 = 1;
```
Comment (mythical-fred) [inline crates/pipeline-manager/src/api/endpoints/pipeline_interaction/support_bundle.rs:57]: Forward-compat win, but `fda debug unbundle` never reads `metadata_version`. When you bump to 2, older `fda` binaries will parse new bundles with whatever guesses they have. Worth a `warn!` in `unbundle_support_bundle` when `metadata_version > known` (or a comment here noting consumers MUST gate).

**Review by mythical-fred — APPROVED** [review]
LGTM. `timestamp_dir` now actually emits `Z` (matches the docstring), and the `test_support_bundle_limit` docstring → comment cleanup is fine. The other open notes from my earlier review remain optional polish (silent-collision `unwrap_or("unknown")` in fda unbundle, missing test for the `limit=0 → 400` arm, `silent_bootstrap` SDK fix without a test) — none worth holding the PR.

— Fred 🐟

**Comment by snkas** [comment]
CI failure:
```
+ file_count=4
+ '[' 4 -ne 2 ']'
+ echo 'Expected 1 file in support bundle (manifest.txt), found 4'
+ unzip -l test-support-bundle-none.zip
Expected 1 file in support bundle (manifest.txt), found 4
Archive:  test-support-bundle-none.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
     6419  2026-05-26 09:31   2026-05-26T09-31-20.319926977Z/pipeline_events.json
      545  2026-05-26 09:31   metadata.txt
      713  2026-05-26 09:31   metadata.json
---------                     -------
     7677                     3 files
```

Is it supposed to be 1, 2, 3, or 4 files?
